BikeRentalApi/endpoints/malfunctionsList.py

```diff
@@ -19,3 +23,49 @@ def get(request):
         safe = False,
         status = status.HTTP_200_OK
     )
+
+
+@RoleRequired([Role.User, Role.Tech, Role.Admin])
+def post(request):
+    user = authenticate_bikes_user(request)
+    stream = io.BytesIO(request.body)
+    data = JSONParser().parse(stream)
+
+    try:
+        id = int(data['id'])
+        description = data['description']
+    except KeyError or ValueError:
+        return HttpResponse(status = status.HTTP_400_BAD_REQUEST)
+
+    bike = Bike.objects.get(id = id)
+
+    if bike is None:
+        return JsonResponse(
+            {'message': 'Bike not found'},
+            status = status.HTTP_404_NOT_FOUND
+        )
+
+    try:
+        rental = Rental.objects.get(bike_id = bike.id)
+    except Rental.DoesNotExist:
+        return JsonResponse(
+            {'message': 'Bike not rented by calling user'},
+            status = status.HTTP_422_UNPROCESSABLE_ENTITY
+        )
+
+    if rental.user.id != user.id:
+        return JsonResponse(
+            {'message': 'Bike not rented by calling user'},
+            status = status.HTTP_422_UNPROCESSABLE_ENTITY
+        )
+
+    malfunction = Malfunction(bike = bike, description = description, reporting_user = user.user)
+    malfunction.save()
```

Comment:
This could be replaced with the following oneliner:
`Malfunction.objects.create(bike = bike, description = description, reporting_user = user.user)`

Reply:
I was under the impression that we've always done it that way (bikesList & stationsList), but now I see that it's mixed through the project. I'll change it, obviously, but also changing it globally in the project might be worth considering.